tests/models/test_main.py

```diff
@@ -565,6 +565,55 @@ def test_query_brief_applications_closed_at_date_for_mix_of_brief_lengths(self):
             db.session.commit()
             assert Brief.query.filter(Brief.applications_closed_at == datetime(2016, 3, 17, 23, 59, 59)).count() == 3
 
+    # TODO: add cases for querying created_at/updated_at auto timestamps with freeze_time
+
+    @pytest.mark.parametrize('inclusive,expected_count', [(True, 2), (False, 1)])
+    @pytest.mark.parametrize('date_format', ["2017-01-01", datetime(2017, 1, 1, 0, 0, 0)])
+    @pytest.mark.parametrize('datestamp_attr', ['published_at', 'withdrawn_at', 'cancelled_at', 'unsuccessful_at'])
+    def test_query_brief_has_date_field_before_and_after(self, datestamp_attr, date_format, inclusive, expected_count):
+        with self.app.app_context():
+            new_briefs = [Brief(data={}, framework=self.framework, lot=self.lot) for i in range(3)]
+            setattr(new_briefs[0], datestamp_attr, datetime(2016, 12, 31, 23, 59, 59))  # < date
+            setattr(new_briefs[1], datestamp_attr, datetime(2017, 1, 1, 0, 0, 0))  # <= and >= date
+            setattr(new_briefs[2], datestamp_attr, datetime(2017, 1, 1, 0, 0, 1))  # > date
+
+            db.session.add_all(new_briefs)
+            db.session.commit()
+
+            briefs_before = Brief.query.has_date_field_before(
+                datestamp_attr, start_date=date_format, inclusive=inclusive
+            )
+            assert briefs_before.count() == expected_count
+
+            briefs_after = Brief.query.has_date_field_after(
+                datestamp_attr, start_date=date_format, inclusive=inclusive
+            )
+            assert briefs_after.count() == expected_count
+
+    @pytest.mark.parametrize('inclusive,expected_count', [(True, 2), (False, 0)])
+    @pytest.mark.parametrize(
+        'start_date,end_date', [
+            ("2017-01-01", "2017-01-03"),
+            (datetime(2017, 1, 1, 0, 0, 0), datetime(2017, 1, 3, 0, 0, 0)),
+        ]
+    )
+    @pytest.mark.parametrize('datestamp_attr', ['published_at', 'withdrawn_at', 'cancelled_at', 'unsuccessful_at'])
+    def test_query_brief_has_date_field_between(self, datestamp_attr, start_date, end_date, inclusive, expected_count):
+        with self.app.app_context():
+            new_briefs = [Brief(data={}, framework=self.framework, lot=self.lot) for i in range(4)]
+            setattr(new_briefs[0], datestamp_attr, datetime(2016, 12, 31, 23, 59, 59))  # < date range
+            setattr(new_briefs[1], datestamp_attr, datetime(2017, 1, 1, 0, 0, 0))  # <= date range
+            setattr(new_briefs[2], datestamp_attr, datetime(2017, 1, 3, 0, 0, 0))  # >= date range
+            setattr(new_briefs[3], datestamp_attr, datetime(2017, 1, 3, 0, 0, 1))  # > date range
```

Comment:
Again, I think this last date is still inside an inclusive date range of 1/1/2017 to 3/1/2017.
A non inclusive range would just be 2/1/17.

It feels like there may be slight confusion/misunderstanding around filtering based on dates rather than datetimes. Feel free to grab me to chat this through if you'd prefer.

Reply:
Yikes, you're right. I had written my docstring wrong on the method itself, and the tests were based on that! Carelessness on my part, sorry.

Reply:
No problem :) Let me know when you want me to take a look after you've made relevant changes.

Reply:
As discussed, the query methods now only accept `datetime` objects and raise a ValueError if a string is supplied. Hopefully this should remove the ambiguity.